pkg/clustersserver/suite_test.go

```diff
@@ -0,0 +1,78 @@
+package clustersserver_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/fluxcd/webui/pkg/clustersserver"
+	"github.com/fluxcd/webui/pkg/rpc/clusters"
+	pb "github.com/fluxcd/webui/pkg/rpc/clusters"
+	"github.com/fluxcd/webui/pkg/util"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	apiruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+)
+
+var testClustername = "test-cluster"
+var testenv *envtest.Environment
+var cfg *rest.Config
+var testclient client.Client
+var err error
+var scheme *apiruntime.Scheme
+var server *httptest.Server
+var c clusters.Clusters
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"clusterserver suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = BeforeSuite(func() {
+	testenv = &envtest.Environment{CRDDirectoryPaths: []string{"../../tools/crd"}}
+
+	cfg, err = testenv.Start()
+
+	Expect(err).NotTo(HaveOccurred())
+
+	scheme = util.CreateScheme()
+})
+
+var _ = BeforeEach(func() {
+	testclient, err = client.New(cfg, client.Options{Scheme: scheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	clusters := clustersserver.Server{
+		AvailableContexts: []string{testClustername},
+		InitialContext:    testClustername,
+		ClientCache:       map[string]client.Client{},
+		CreateClient: func(kubeContext string) (client.Client, error) {
```

Comment:
You can probably get a kubecontext for the testenv, if you want to test that client creation works. But may not be relevant here.

Reply:
Not sure what you mean @squaremo, can you clarify?

Reply:
You could implement clusterserver.Server in such a way that you just point it at a kubeconfig file, and `InitialContext` etc. were implemented in terms of what's in the file, rather than all having to be supplied. At present it's possible to create a `clusterserver.Server` that isn't internally consistent; e.g., the initial context isn't in the available contexts.

However: it's not something you're trying to solve here, so feel free to disregard.

Reply:
I didn't want to rely on having file system access. `CreateClient` is only exported for these tests.